src/NuGet.CommandLine/MultipartWebRequest.cs

```diff
@@ -0,0 +1,124 @@
+
+        private readonly List<PostFileData> _files;
+
+        public MultipartWebRequest()
+            : this(new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase))
+        {
+        }
+
+        public MultipartWebRequest(Dictionary<string, string> formData)
+        {
+            _formData = formData;
+            _files = new List<PostFileData>();
+        }
+
+        public void AddFormData(string key, string value)
+        {
+            _formData.Add(key, value);
+        }
+
+        public void AddFile(Func<Stream> fileFactory, string fieldName, long length, string contentType = "application/octet-stream")
+        {
+            _files.Add(new PostFileData
+            {
+                FileFactory = fileFactory,
+                FieldName = fieldName,
+                ContentType = contentType,
+                ContentLength = length
+            });
+        }
+
+        public void CreateMultipartRequest(WebRequest request)
+        {
+            string boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x", CultureInfo.InvariantCulture);
+            request.ContentType = "multipart/form-data; boundary=" + boundary;
+            request.ContentLength = CalculateContentLength(boundary);
+
+            using (Stream stream = request.GetRequestStream())
+            {
+                foreach (var item in _formData)
+                {
+                    string header = String.Format(CultureInfo.InvariantCulture, FormDataTemplate, boundary, item.Key, item.Value);
+                    byte[] headerBytes = Encoding.UTF8.GetBytes(header);
+                    stream.Write(headerBytes, 0, headerBytes.Length);
+                }
+
+                byte[] newlineBytes = Encoding.UTF8.GetBytes("\r\n");
+                foreach (var file in _files)
+                {
+                    string header = String.Format(CultureInfo.InvariantCulture, FileTemplate, boundary, file.FieldName, file.FieldName, file.ContentType);
+                    byte[] headerBytes = Encoding.UTF8.GetBytes(header);
+                    stream.Write(headerBytes, 0, headerBytes.Length);
+
+                    Stream fileStream = file.FileFactory();
+                    fileStream.CopyTo(stream, bufferSize: 4 * 1024);
+                    fileStream.Close();
+                    stream.Write(newlineBytes, 0, newlineBytes.Length);
+                }
+
+                string trailer = String.Format(CultureInfo.InvariantCulture, "--{0}--", boundary);
+                byte[] trailerBytes = Encoding.UTF8.GetBytes(trailer);
+                stream.Write(trailerBytes, 0, trailerBytes.Length);
+            }
+        }
+
+        private long CalculateContentLength(string boundary)
+        {
+            long totalContentLength = 0;
+
+            foreach (var item in _formData)
+            {
+                string header = String.Format(CultureInfo.InvariantCulture, FormDataTemplate, boundary, item.Key, item.Value);
```

Comment:
Use smaller case string for the static method string.Format. Please do the same for all other uses of static methods of string.